core/templates/dev/head/exploration_editor/ExplorationEditor.js

```diff
@@ -923,17 +923,7 @@ oppia.controller('ExplorationSaveAndPublishButtons', [
 
         modalInstance.result.then(function(commitMessage) {
           _modalIsOpen = false;
-          saveDraftToBackend(commitMessage, function() {
-            explorationData.getLastSavedData().then(function(data) {
-              explorationData.getData().then(function(currentData) {
-                if (data.version > currentData.version) {
-                  autosaveInfoModalsService.showVersionMismatchModal(
-                    changeListService.getChangeList());
-                  return;
-                }
-              });
-            });
-          });
+          saveDraftToBackend(commitMessage);
```

Comment:
@jacobdavis11  Hmm. `autosaveChangeListOnChange` and `saveDraftToBackend` have different utilities, as in - the former hits `/createhandler/autosave_draft/<id>` and returns whether the version of draft is valid (`is_version_of_draft_valid` )in the payload whereas the latter does return to the frontend whether the version of draft is valid or not, (although it throws an exception if the version is not valid!). Are you suggesting that the version validity be returned to frontend even in case of the `saveDraftToBackend` call as well instead of throwing an exception -- so that the relevant modals can be opened ?